### PR TITLE
Documentation michelson encoder

### PR DESCRIPTION
**Pull request by roxaneletourneau**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: docs/michelson_encoder.md [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+- `ParameterSchema`
+- `Schema`
+
+[Here](/img/taquito-michelson-encoder.png) is a class diagram showing their members, methods, and relations.
+
+A specific token class in the package represents each different Michelson type (i.e., `nat`, `list`, `pair`, `or`, ...). Each of these classes extends the abstract class `Token` and needs to implement these four inherited abstract methods:
+- `Execute`: To convert Michelson data into familiar-looking javascript data
+- `Encode`: To convert javascript array data to JSON Michelson
+- `EncodeObject`: To convert javascript object data to JSON Michelson
+- `ExtractSchema`: To shows how the data should be structured in the javascript format
```
Comment (michaelkernaghan) [inline docs/michelson_encoder.md:23]: "to show how ...

File: docs/michelson_encoder.md [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+
+We can reason about Michelson type and data as tree structures. Thus, the methods of the `Token` class use recursion to iterate over a Michelson parameter and create a specific token for each Michelson type encountered (i.e., each node and leaf of the tree).
```
Comment (michaelkernaghan) [inline docs/michelson_encoder.md:25]: about Michelson types ...

File: docs/michelson_encoder.md [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+
+The constructors of the `ParameterSchema` and `Schema` classes take a `MichelsonV1Expression` has a parameter (i.e., a Michelson type in JSON format). These two classes have a composition relationship with the `Token` class as they have a private member named `root`, which is a `Token` instance corresponding to the root type of the Michelson parameter. The root is the starting point of the recursive calls.
```
Comment (michaelkernaghan) [inline docs/michelson_encoder.md:27]: ...  `MichelsonV1Expression` as a parameter ...               (not has)

File: docs/michelson_encoder.md [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+
+The `Schema` class is internally used in Taquito:
+- When calling `Tezos.contract.getStorage()`:
+    It allows returning a well-formatted JSON object of the contract storage using the `Execute` method to convert the Michelson data into familiar-looking javascript data.
+- When fetching a bigmap key with `BigMapAbstraction.get()` or `RpcContractProvider.getBigMapKey()`:
+    It transforms the key we want to fetch into Michelson using the `EncodeBigMapKey` method, and it transforms the fetched value into a javascript object using the `ExecuteOnBigMapValue`.
+- When deploying a contract:
+    The `Encode` method allows transforming the javascript object used for the storage into Michelson data.
+- In the tzip16 package:
+    The `FindFirstInTopLevelPair` method allows finding a value in the storage matching a specified type (in this case, the big map named metadata)
```
Comment (michaelkernaghan) [inline docs/michelson_encoder.md:404]: end the sentence with a period

**Comment by michaelkernaghan** [comment]
I checked all the live code examples in the preview and they all work correctly.

**Comment by roxaneletourneau** [comment]
> I checked all the live code examples in the preview and they all work correctly.

Thank you @michaelkernaghan ! I corrected the typos you reported.

**Review by kjaiswal — APPROVED** [review]

**Review by jevonearth — APPROVED** [review]

**Pull request merged**